plugins/servicenow/icon_servicenow/connection/connection.py

```
@@ -29,7 +29,7 @@ def connect(self, params):
         oauth_client_id = params.get(Input.CLIENT_ID)
         oauth_client_secret = params.get(Input.CLIENT_SECRET, {}).get("secretKey")
 
-        if not oauth_client_id or not oauth_client_secret:
+        if not oauth_client_id or not oauth_client_secret and username and password:
```

I think we should add case where none of `username`, `password`, `client_id` and `client_secret` are provided. In that case we should raise an exception that something has to be entered.

---

make sense, it's covered now @igorski-r7 , thanks